ops_boba/monitor/exec/run-monitor.js

```diff
@@ -1,77 +1,181 @@
-const stateRootMonitorService = require('../services/stateRootMonitor');
-const exitMonitorService = require('../services/exitMonitor');
-const l1BridgeMonitorService = require('../services/l1BridgeMonitor');
-const messageMonitorService = require('../services/messageMonitor');
-const { sleep } = require('@eth-optimism/core-utils');
-const { setupProvider, validateMonitoring } = require('../services/monitoring');
-const { logger } = require('../services/utilities/logger');
-const configs = require('../services/utilities/configs');
+const mysql = require('mysql')
+const util = require('util')
+const configs = require('../services/utilities/configs')
+const { sleep } = require('@eth-optimism/core-utils')
+const { logger } = require('../services/utilities/logger')
 
 const loop = async (func) => {
   while (true) {
     try {
-      await func();
+      await func()
     } catch (error) {
       console.log('Unhandled exception during monitor service', {
         message: error.toString(),
         stack: error.stack,
         code: error.code,
-      });
-      await sleep(1000);
+      })
+      await sleep(1000)
     }
   }
 }
 
+const loopLogTx = async () => {
+  while (true) {
+    try {
+      await getLatestTx()
+    } catch (error) {
+      logger.error('Unhandled exception during logging tx', {
+        message: error.toString(),
+        stack: error.stack,
+        code: error.code,
+      })
+    }
+    try {
+      await getLatestReceipt()
+    } catch (error) {
+      logger.error('Unhandled exception during logging receipt', {
+        message: error.toString(),
+        stack: error.stack,
+        code: error.code,
+      })
+    }
+    await sleep(5000)
+  }
+}
+
+const getLatestTx = async () => {
+  const con = mysql.createConnection({
+    host: configs.rdsEndpoint,
+    port: configs.rdsPort,
+    user: configs.rdsMysqlName,
+    password: configs.rdsMysqlPassword,
+  })
+
+  const query = util.promisify(con.query).bind(con)
+  await query(`USE ${configs.rdsDbNameTx}`)
+  const txs = await query(
+    `SELECT * FROM tx WHERE startTime>${configs.startTimeLog} ORDER BY startTime LIMIT 10`
+  )
+
+  if (txs.length > 0) {
+    txs.forEach((tx, index) => {
+      if (index === txs.length - 1) {
+        configs.startTimeLog = `${tx.startTime}`
+      }
+      tx.startTime = parseInt(`${tx.startTime}`, 10)
+      tx.endTime = parseInt(`${tx.endTime}`, 10)
+      tx.duration = tx.endTime - tx.startTime
+      tx.endISOTime = new Date(parseInt(`${tx.endTime}`, 10)).toISOString()
+      tx.startISOTime = new Date(parseInt(`${tx.startTime}`, 10)).toISOString()
+      logger.info('tx', tx)
+    })
+  }
+  con.end()
+}
+
+const getLatestReceipt = async () => {
+  const con = mysql.createConnection({
+    host: configs.rdsEndpoint,
+    port: configs.rdsPort,
+    user: configs.rdsMysqlName,
+    password: configs.rdsMysqlPassword,
+  })
+
+  const query = util.promisify(con.query).bind(con)
+  await query(`USE ${configs.rdsDbNameReceipt}`)
```

Comment:
Could you please move the database requests to database.service.js?

Reply:
@cby3149 I've moved the database requests to database.service.js. Please take a look. Thanks